MoreLinq/RandomSubset.cs

```diff
@@ -76,8 +76,11 @@ static IEnumerable<T> RandomSubsetImpl<T>(IEnumerable<T> source, Random rand, Fu
 
             if (array.Length < subsetSize)
             {
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly
+                               // TODO Throw InvalidOperationException instead?
```

Comment:
`ArgumentOutOfRangeException` is valid here. CA2208 is a good check, but cannot be coded to evaluate this scenario. disabling it for this line is the most correct approach, imo

Reply:
Do you know of LINQ operators from the BCL that do argument validate at time of iteration/evaluation?

Reply:
No, but none of the BCL LINQ operators have any arguments that could be invalid, but are only known to be invalid at iteration.

Reply:
It can also be left as undefined behaviour then. I don't think `ArgumentOutOfRangeException` is the right call as it could be triggered far from the call site.

Reply:
For now I think we can close this and revisit when the to-do comment is being address because there's no pressing need to make a binary breaking change here. It's been this way for a while.